syntax = "proto3";

package bookstore.order_queue;

// Minimal order structure for the queue
message QueueOrder {
  string orderId = 1;
  string shippingCountry = 2;
  string jsonData = 3; 
}

// Request for enqueuing
message EnqueueRequest {
  QueueOrder order = 1;
}

// Response for enqueuing
message EnqueueResponse {
  bool success = 1;
  string message = 2;
}

// Request for dequeue
message DequeueRequest {}

// Response for dequeue
message DequeueResponse {
  bool success = 1;
  string message = 2;
  QueueOrder order = 3;
}

// The service: enqueue + dequeue
service OrderQueueService {
  rpc Enqueue (EnqueueRequest) returns (EnqueueResponse);
  rpc Dequeue (DequeueRequest) returns (DequeueResponse);
}
